addons/website_mass_mailing/controllers/main.py

```diff
@@ -9,42 +9,47 @@
 class MassMailController(main.MassMailController):
 
     @route('/website_mass_mailing/is_subscriber', type='json', website=True, auth="public")
-    def is_subscriber(self, list_id, **post):
-        email = None
-        if not request.env.user._is_public():
-            email = request.env.user.email
-        elif request.session.get('mass_mailing_email'):
-            email = request.session['mass_mailing_email']
+    def is_subscriber(self, list_id, input_name, **post):
+        value = self.set_default_input_value(input_name)
 
         is_subscriber = False
-        if email:
-            contacts_count = request.env['mailing.contact.subscription'].sudo().search_count([('list_id', 'in', [int(list_id)]), ('contact_id.email', '=', email), ('opt_out', '=', False)])
+        if value:
+            contacts_count = request.env['mailing.contact.subscription'].sudo().search_count([('list_id', 'in', [int(list_id)]), ('contact_id.' + input_name, '=', value), ('opt_out', '=', False)])
```

Comment:
Composing domains based on arbitrary input is dangerous. I suppose this is done for sms override. I would have the ``set_default_input_value`` also returns the field name to check (email or mobile), so that we don't do that kind of composition.